android/src/toga_android/window.py

```diff
@@ -137,8 +144,73 @@ def get_visible(self):
     # Window state
     ######################################################################
 
-    def set_full_screen(self, is_full_screen):
-        self.interface.factory.not_implemented("Window.set_full_screen()")
+    def get_window_state(self):
+        # window.state is called in _close(), which itself sometimes
+        # is called during early stages of app startup, during which
+        # the app attribute may not exist. In such cases, return NORMAL.
```

Comment:
I agree I was wrong. It should have been worded as "during the setup of a new test after completion of a previous test". 

if I modify `get_window_state()` as:
```
def get_window_state(self):
    # window.state is called in _close(), which itself sometimes
    # is called during early stages of app startup, during which
    # the app attribute may not exist. In such cases, return NORMAL.

    if getattr(self, "app", None) is None:
        raise RuntimeError

    decor_view = self.app.native.getWindow().getDecorView()
    system_ui_flags = decor_view.getSystemUiVisibility()
    ....
```

Then on running the testbed on Android, every test after `tests/window/test_window.py::test_secondary_window` fails:
```
I/python.stdout: ============================= test session starts ==============================
I/python.stdout: collecting ...
I/python.stdout: collected 12 items
I/python.stdout:
I/python.stdout: tests/window/test_window.py::test_title
I/python.stdout: PASSED                           [  8%]
I/python.stdout:
I/python.stdout: tests/window/test_window.py::test_visibility
I/python.stdout: PASSED                      [ 16%]
I/python.stdout:
I/python.stdout: tests/window/test_window.py::test_secondary_window
I/python.stdout: PASSED                [ 25%]
I/python.stdout:
I/python.stdout: tests/window/test_window.py::test_move_and_resize
I/python.stdout: ERROR                  [ 33%]
I/python.stdout:
I/python.stdout: ...
```
Stack dump of the test failure:
```
I/python.stdout: ____________________ ERROR at setup of test_move_and_resize ____________________
I/python.stdout: Traceback (most recent call last):
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/runner.py", line 341, in from_call
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/runner.py", line 241, in <lambda>
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_hooks.py", line 513, in __call__
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_manager.py", line 120, in _hookexec
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_callers.py", line 139, in _multicall
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_callers.py", line 122, in _multicall
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/unraisableexception.py", line 85, in pytest_runtest_setup
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/unraisableexception.py", line 65, in unraisable_exception_runtest_hook
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_callers.py", line 122, in _multicall
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/logging.py", line 844, in pytest_runtest_setup
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/logging.py", line 833, in _runtest_for
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_callers.py", line 122, in _multicall
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/capture.py", line 873, in pytest_runtest_setup
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_callers.py", line 122, in _multicall
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/threadexception.py", line 82, in pytest_runtest_setup
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/threadexception.py", line 63, in thread_exception_runtest_hook
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_callers.py", line 103, in _multicall
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/runner.py", line 159, in pytest_runtest_setup
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/runner.py", line 517, in setup
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/runner.py", line 514, in setup
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/python.py", line 1635, in setup
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/fixtures.py", line 686, in _fillfixtures
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/fixtures.py", line 521, in getfixturevalue
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/fixtures.py", line 606, in _get_active_fixturedef
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/fixtures.py", line 1076, in execute
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_hooks.py", line 513, in __call__
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_manager.py", line 120, in _hookexec
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_callers.py", line 182, in _multicall
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_result.py", line 100, in get_result
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_callers.py", line 167, in _multicall
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/setuponly.py", line 36, in pytest_fixture_setup
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pluggy/_callers.py", line 103, in _multicall
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/fixtures.py", line 1125, in pytest_fixture_setup
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/_pytest/fixtures.py", line 887, in call_fixture_func
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pytest_asyncio/plugin.py", line 366, in _async_fixture_wrapper
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/app/tests/conftest.py", line 130, in run_until_complete
I/python.stdout:     return asyncio.run_coroutine_threadsafe(coro, self.loop).result()
I/python.stdout:   File "stdlib/concurrent/futures/_base.py", line 458, in result
I/python.stdout:   File "stdlib/concurrent/futures/_base.py", line 403, in __get_result
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/pytest_asyncio/plugin.py", line 361, in setup
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/app/tests/conftest.py", line 73, in window_cleanup
I/python.stdout:     window.close()
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/toga/window.py", line 315, in close
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/toga/window.py", line 325, in _close
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/toga/window.py", line 471, in state
I/python.stdout:   File "/data/data/org.beeware.toga.testbed/files/chaquopy/AssetFinder/requirements/toga_android/window.py", line 152, in get_window_state
I/python.stdout:     raise RuntimeError
I/python.stdout: RuntimeError
```
This indicates that, the  conftest is causing the expected call to `window.close() `after each test:
https://github.com/beeware/toga/blob/d7228ba9255a0314bd2487a26dc97aac5006120f/testbed/tests/conftest.py#L60-L78

This cleanup behavior is expected, but I am a bit surprised that `window.close()` actually calls `_close()` on Android, since the same `MainWindow `is used for all the tests.

But, since `get_window_state()` encounters a situation, where `self.app` is not assigned, this check guard is needed.